src/xbot/common/controls/sensors/XXboxController.java

```diff
@@ -48,6 +56,16 @@ public int getValue() {
     }
     
     public AdvancedXboxButton getXboxButton(XboxButton buttonName) {
+        
+        if (!allocatedButtons.containsKey(buttonName)) {
+            allocatedButtons.put(buttonName, true);
```

Comment:
Could we instead maintain a map of `AdvancedXboxButton`s which contains all the buttons we've returned and check against that? It feels odd to be returning a button which we lose all references to and instead maintain state that describes whether we've ever generated a given button. I'm thinking that keeping references to these buttons might be useful in the future.